internal_ws/ykpack/src/types.rs

```diff
@@ -132,6 +132,9 @@ impl Ty {
         }
     }
 
+    // XXX this function feels a bit evil. surely it should be the caller's job to do this? that
+    // way the need to panic will either a) be clear b) not be possible because the caller has
+    // verified things?
```

Comment:
`unwrap_*` is a commonly used name for a function to call when only a certain variant should be passed, or it would indicate a bug.

Reply:
Yep. It's just a convenience function to cut down on code duplication.

Reply:
OK. At least it needs a `# Safety` or `#Panics` qualification in the doc.

Reply:
This is new to me. Surely every function that may panic doesn't need a special doc?

Reply:
It shouldn't be news to you: the Rust std lib documents this carefully :) If you think about it, it's really another way of saying "what are this function's preconditions?" We don't have to document *unexpected* panics, but we should document the expected panics IMHO.

Reply:
A panic here would indeed be unexpected. It's not in a public API :P

Reply:
By "unexpected" meaning "if this panics, I, the function, have got something wrong" vs. "expected" meaning "you the caller did not respect the precondition I require you to respect." The precondition in the latter thus needs documenting.

Reply:
I have no issue documenting it, but i'm surprised it's necessary. This idiom is really common :P

Reply:
I thought the same at one point, but then realised that if we *always* do it, we never have to make the judgment call -- and in the long run, it saves time for me not to think about it.

Reply:
https://github.com/softdevteam/yk/pull/261